src/rgw/rgw_aws_sns.cc

```diff
@@ -0,0 +1,86 @@
+#include <boost/lockfree/queue.hpp>
+#include <string>
+#include <utility>
+#include <aws/sns/SNSClient.h>
+#include <aws/sns/model/PublishRequest.h>
+
+namespace sns {
+
+    static const int STATUS_OK = 1;
+    static const int STATUS_FULL = 2;
+    static const int STATUS_MANAGER_STOPPED = 3;
+
+    struct message_t {
+        std::string topic;
+        std::string message;
+
+        message_t(std::string topic, std::string message) : topic(std::move(topic)), message(std::move(message)) {};
+    };
+
+    typedef boost::lockfree::queue<message_t *, boost::lockfree::fixed_sized<true>> MessageQueue;
+
+    class Manager {
+    private:
+        MessageQueue messageQueue;
+        std::thread runner;
+        bool stopped;
+
+        void publish_internal(message_t *msg) {
+          std::cout << "Publishing.." << std::endl;
+          Aws::SNS::Model::PublishRequest req;
+          req.SetMessage(Aws::String(msg->message));
+          req.SetTopicArn(Aws::String(msg->topic));
+          Aws::SNS::SNSClient client;
```

Comment:
client could be a data member of Manager, instead of constructing it for every message

Reply:
where do you set the client parameters?
which parameters should the client expect?